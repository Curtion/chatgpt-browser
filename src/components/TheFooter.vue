<script setup lang="ts">
let modelValue = $ref('')
function handleSendMsg() {
  if (!modelValue)
    return
  sendMsg(modelValue)
  modelValue = ''
}

function handleDelMsg() {

}
</script>

<template>
  <div
    w-full
    px-1
    sm:w-screen-sm
    sm:px-0
    md:w-screen-md
    md:px-0
    lg:w-screen-lg
    lg:px-0
    m-auto
    h-12
    flex
    justify-center
    items-center
    gap-2
  >
    <div text-xl hover:cursor-pointer i-carbon-trash-can @click="handleDelMsg" />
    <input
      id="input"
      v-model="modelValue"
      type="text"
      p="x-4"
      bg="transparent"
      w="full"
      h="full"
      max-w-screen-lg
      border="~ rounded gray-200 dark:gray-700"
      outline="none active:none"
    >
    <div text-xl hover:cursor-pointer i-carbon-send @click="handleSendMsg" />
  </div>
  <div h-8 mt-2 shadow flex justify-center items-center text-xs>
    ©竹影流浪 2014-2023
  </div>
</template>
