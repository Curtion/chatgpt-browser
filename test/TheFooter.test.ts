import { mount } from '@vue/test-utils'
import { describe, expect, it } from 'vitest'
import TheFooter from '../src/components/TheFooter.vue'

describe('TheFooter.vue', () => {
  it('should render', () => {
    const wrapper = mount(TheFooter)
    expect(wrapper.text()).toContain('2023')
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should be interactive', async () => {
    const wrapper = mount(TheFooter)
    expect(wrapper.text()).toContain('2023')

    expect(wrapper.find('.inc').exists()).toBe(true)

    await wrapper.get('button').trigger('click')

    expect(wrapper.text()).toContain('1')
  })
})
