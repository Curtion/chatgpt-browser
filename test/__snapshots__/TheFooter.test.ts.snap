// Vitest Snapshot v1

exports[`TheFooter.vue > should render 1`] = `
"<div w-full=\\"\\" px-1=\\"\\" sm:w-screen-sm=\\"\\" sm:px-0=\\"\\" md:w-screen-md=\\"\\" md:px-0=\\"\\" lg:w-screen-lg=\\"\\" lg:px-0=\\"\\" m-auto=\\"\\" h-12=\\"\\" flex=\\"\\" justify-center=\\"\\" items-center=\\"\\" gap-2=\\"\\">
  <div text-xl=\\"\\" hover:cursor-pointer=\\"\\" i-carbon-trash-can=\\"\\" class=\\"hover:cursor-pointer\\"></div><input id=\\"input\\" type=\\"text\\" p=\\"x-4\\" bg=\\"transparent\\" w=\\"full\\" h=\\"full\\" max-w-screen-lg=\\"\\" border=\\"~ rounded gray-200 dark:gray-700\\" outline=\\"none active:none\\" placeholder=\\"请输入内容\\">
  <div text-xl=\\"\\" class=\\"i-carbon-send\\" hover:cursor-pointer=\\"\\"></div>
</div>
<div h-8=\\"\\" mt-2=\\"\\" shadow=\\"\\" flex=\\"\\" justify-center=\\"\\" items-center=\\"\\" text-xs=\\"\\"> ©竹影流浪 2014-2023 </div>"
`;
